docs/plugin-development/interacting-with-the-game.md

```diff
@@ -0,0 +1,269 @@
+
+Most plugins will stay firmly in the realm of stages 1 and 2, with stage 3 being used for novel concepts that have yet
+to be [reverse engineered](reverse-engineering.md) or fully understood. Where possible, plugin developers who do
+reverse engineering as part of their plugin development are encouraged to contribute their findings back to the Client
+Structs project, so that other developers may use them in the future.
+
+This document page won't explain how to use Dalamud-provided APIs to interact with the game, as they're all otherwise
+documented and hopefully accessible. Instead, this document will focus on the more advanced concepts; that is, where 
+the Dalamud API doesn't quite reach.
+
+## I Want To Do That!
+
+Sometimes, it is beneficial to ask the game itself to do something, rather than doing it yourself. In effect, this 
+means using the game code as a library where any arbitrary function can be called and their results used freely. This
+allows plugins to perform calculations in the same way the game does, or otherwise take actions in the game just as 
+it would if Dalamud weren't even there.
+
+For example, a plugin might want to check if the player is a mentor:
+
+```csharp
+public unsafe bool IsPlayerMentor() {
+    var playerStatePtr = PlayerState.Instance();
+    return playerStatePtr->IsMentor();
+}
+```
+
+This method will grab the instance of `PlayerState` from Client Structs, and call the appropriate check. 
+
+### Making Your Own Delegates
+
+Sometimes, a method you're interested in might not be in Client Structs. When this happens, a developer can engage
+their reverse engineering prowess to generate a signature, which they can then use to create their own delegate:
+
+```c#
+public class GameFunctions {
+    private delegate byte IsQuestCompletedDelegate(ushort questId);
+
+    [Signature("E8 ?? ?? ?? ?? 41 88 84 2C")]
+    private readonly IsQuestCompletedDelegate? _isQuestCompleted = null;
+
+    public GameFunctions() {
+        SignatureHelper.Initialise(this);
+    }
+  
+    public bool IsQuestCompleted(ushort questId) {
+        if (this._isQuestCompleted == null) 
+            throw new InvalidOperationException("IsQuestCompleted signature wasn't found!");
+      
+        return this._isQuestCompleted(questId) > 0;
+  }
+}
+```
+
+This is a lot of code, so let's break it down a bit.
+
+First, the developer declares a [delegate][delegate-doc] for the function they want to call. This informs the compiler
+and the code of the return type (in this case, a `byte`), as well as the arguments of the function. This line alone is
+purely declaratory, and has no impact other than definition. If a specific argument is a reference to an undocumented
+pointer (or the developer simply doesn't care about accessing any data inside the struct target), the `nint` type will
+often be used.
+
+Next, the developer declares a nullable *instance* of that delegate, with its default value set to `null`. This 
```

Comment:
Explain nullabillity here and the ramifications of this choice - if not null, the signature will be treated as `Infallible` and throw an error if it's not found. Provide additional context as to when and why a developer should use a fallible vs an infallible signature.